### DISCUSSION TITLE
Improving OpenAPI spec generation: automatically adding links and non-successful response codes

### DISCUSSION THREAD

**Discussion by shermari** [encode/django-rest-framework#9114] [Ideas & Suggestions]
We've been using DRF for quite some time now and it's been great to use it.
We rely heavily on the OpenAPI specification that's generated (https://www.django-rest-framework.org/community/3.10-announcement/#openapi-schema-generation). Our front end applications consume the OpenAPI spec in order to interact with our DRF back end. This is all working as expected.

Lately though, we've been investing in test-automation and found the OpenAPI spec  to be a great source for automatically generating test-scenario's. However, we noticed that DRF's current implementation of the OpenAPI spec seems to have some limitations that keep us from fully leveraging its potential for test-automation.

We notice 2 missing parts in the OpenAPI spec generated by DRF, that would benefit test-automation and improve the OpenAPI spec in general:
1. Links are not generated. OpenAPI supports links between endpoints (https://swagger.io/docs/specification/links/). I was expecting schema generation to extract that information from foreign key relationships but I couldn’t find anything which will generate such output
2. Only ' successful'  response codes (200 etc.) are mentioned in the OpenAPI spec, while the API could also return non-successful response codes (404 etc. see: https://spec.openapis.org/oas/latest.html#responses-object)

We'd like to know if 1. automatically adding links and 2. automatically adding non-successful response codes to the OpenAPI spec is currently supported. 

If not, we'd like to raise a feature request, since this doesn't add a new feature DRF (which is considered feature complete) but simply improves an existing feature by generating a more complete OpenAPI spec, that follows what OpenAPI 3.x already offers.

Environment information:
python version: 3.9.6
drf version: 3.12.4
django version: 3.1.7
schemathesis version:3.19.6
OS: Mac OS Monterey, 12.0.1

**Comment by auvipy** [comment]
using https://drf-spectacular.readthedocs.io/en/latest/ would be recommended instead of using the built in one for OpenAPI. you can try it instead.

**Reply by shermari** [reply]
@auvipy  Thank you for response to my query.

I tried the drf-spectacular (https://drf-spectacular.readthedocs.io/en/latest/ ) as per your suggestion, but unfortunately it does not made any difference. 
In order to clarify I would say I searched the whole web for answers but couldn't find any example that how to generate openapi links (https://swagger.io/docs/specification/links/)
I would really appreciate if if any one has any idea that how can we generate links programmatically.
I know I can manually add these links to the openapi YAML file but that will take me lot of time and difficult to maintain as we have few hundred endpoints where we need to add links.
